tox.ini

```diff
@@ -1,37 +1,75 @@
 [tox]
 description = Default tox environments list
 envlist =
-    style,{py38,py39,py310,py311,py312}{,-coverage},doc
+    code-style
+    doc-style
+    {py39,py310,py311,py312}{,-coverage}
+    doc-{links,html}
+    docker-run-{linux_host,windows_host}
 skip_missing_interpreters = true
 isolated_build = true
 isolated_build_env = build
-
 [testenv]
 description = Checks for project unit tests and coverage (if desired)
 basepython =
-    py39: python3.9
-    py310: python3.10
-    py311: python3.11
-    py312: python3.12
     py: python3
-    {style,reformat,doc}: python3
+    {code-style,doc-links,doc-html, tests}: python3
 setenv =
     PYTHONUNBUFFERED = yes
     coverage: PYTEST_EXTRA_ARGS = --cov=ansys.mapdl --cov-report=term --cov-report=xml:.cov/xml --cov-report=html:.cov/html
 extras = tests
 commands =
     pytest {env:PYTEST_MARKERS:} {env:PYTEST_EXTRA_ARGS:} {posargs:-vv}
 
-[testenv:style]
+[testenv:code-style]
 description = Checks project code style
+skip_install = true
 deps =
     pre-commit
 commands =
     pre-commit install
     pre-commit run --all-files --show-diff-on-failure
 
-[testenv:doc]
+[testenv:doc-style]
+description = Checks project documentation style
+skip_install = true
+allowlist_externals =
+    vale
+commands =
+    vale sync --config="{toxinidir}/doc/.vale.ini"
+    vale --config="{toxinidir}/doc/.vale.ini" "{toxinidir}/doc" 
+
+[testenv:doc-{links, html}]
 description = Check if documentation generates properly
+setenv =
+    links: BUILDER = linkcheck
+    html: BUILDER = html
+passenv =
+    BUILD_API
+    BUILD_EXAMPLES
+    ON_DOCUMENTATION
 extras = doc
 commands =
-    sphinx-build -d "{toxworkdir}/doc_doctree" doc/source "{toxworkdir}/doc_out" --color -vW -bhtml
\ No newline at end of file
+    sphinx-build -d "{toxworkdir}/doc_doctree" doc/source "{toxinidir}/doc/_build/{env:BUILDER}" --color -vW -b {env:BUILDER} -j auto
+
+[testenv:docker-run-{linux_host, windows_host}]
+description = Run tests in a docker container
+skip_install = true
+allowlist_externals =
+    docker
+platform = 
+    linux_host: linux
+    windows_host: win32
+passenv = 
+    ANSYSLMD_LICENSE_FILE
+    LOCAL_MAPDL_PORT
+    MAPDL_DOCKER_REGISTRY_URL
+    GITHUB_USERNAME
+    GITHUB_TOKEN
+commands =                                                                                                            
+    docker login ghcr.io -u {env:GITHUB_USERNAME} -p {env:GITHUB_TOKEN}
+    docker run \
+    --env  ANSYSLMD_LICENSE_FILE={env:ANSYSLMD_LICENSE_FILE} \
+    --name mapdl -p {env:LOCAL_MAPDL_PORT}:50052 \
```

Comment:
Thanks for the heads up, @germa89.

We should take advantage of this and add a rule for building a docker image locally. This will help us to review the `docker/` directory and the current image building process.

This directory could look like this:

```text
docker/
|- linux/
   |- Dockerfile
   |- docker-compose.yml
   |- distributions/
      |- .gitignore  # Ignore everything except the dir itself
      |- .gitkeep
```

Users should just copy MAPDL artifacts in the `distributions/` dir. Then, running:

`tox -e docker-build-images` would trigger the `docker-compose.yml` file.

Reply:
Building the docker image is a bit complex.. because the ``dockerignore`` depends on the MAPDL version.
The ``docker`` directory is not used by the CICD nor myself to build the docker image at the moment, hence it is not fully maintained.
We have some instructions about building the docker image in: https://mapdl.docs.pyansys.com/version/stable/getting_started/make_container.html 

We could meet next week to discuss this more in details.